internal/utils/chaincode.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package utils
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"github.com/hyperledger/fabric-protos-go/peer/lifecycle"
+	"github.com/hyperledger/fabric/protoutil"
+)
+
+const MrEnclaveLength = 32
+
+func GetChaincodeDefinition(chaincodeId string, stub shim.ChaincodeStubInterface) (*lifecycle.QueryApprovedChaincodeDefinitionResult, error) {
+	function := "QueryApprovedChaincodeDefinition"
+	args := &lifecycle.QueryApprovedChaincodeDefinitionArgs{
+		Name:     chaincodeId,
+		Sequence: 0, // TODO get current sequence number
```

Comment:
good catch! this needs to be updated

Reply:
just double checked with fabric docu. see https://hyperledger-fabric.readthedocs.io/en/latest/commands/peerlifecycle.html#peer-lifecycle-chaincode-querycommitted

Reply:
@mbrandenburger @bvavala I'm a bit confused here: Bruno raises two good points but the merged seems to address only the first one but not the second. Looking at the link marcus sent i don't see how how that would address it. Was that the wrong link and you want to point to peer lifecycle chaincode queryapproved? But that link seems rather to confirm that Bruno has said . So with the current merged code still calling QueryApprovedChaincodeDefinition i am not sure this is really what we want. On the one hand it would seem to mean that retrieval might fail for peers which didn't approve even if the chaincode was committed but on the other hand it also could return something which is not committed?

Reply:
Here is the [QueryChaincodeDefinition](https://github.com/hyperledger/fabric/blob/17f5c7fb020a9228af8791e484783f6a8c5b6da2/core/chaincode/lifecycle/lifecycle.go#L622).

Michael, you're right. Marcus has correctly updated the protobufs, and the update of the most important line went  unnoticed. Surprisingly, the protobufs are similar enough that they do not trigger a type-check error.
I am going to fix it right away.